resolwe/elastic/indices.py

```diff
@@ -0,0 +1,200 @@
+""".. Ignore pydocstyle D400.
+
+===============
+Elastic Indices
+===============
+
+Main two classes
+
+
+.. autoclass:: resolwe.elastic.indices.BaseDocument
+    :members:
+
+.. autoclass:: resolwe.elastic.indices.BaseIndex
+    :members:
+
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import elasticsearch_dsl as dsl
+
+from guardian.models import GroupObjectPermission, UserObjectPermission
+
+from resolwe.flow.models import dict_dot
+
+
+__all__ = ('BaseDocument', 'BaseIndex')
+
+
+class BaseDocument(dsl.DocType):
+    """Base document class to build ElasticSearch documents.
+
+    This is standard ``elasticsearch-dsl`` ``DocType`` class with
+    already added fields for handling permissions.
+
+    """
+
+    #: list of user ids with view permission on the object
+    users_with_permissions = dsl.String()
```

Comment:
Shouldn't there be a `multi=True` here as it accepts an array of strings (also below)?